server/routes/auth-routes.js

```diff
@@ -0,0 +1,85 @@
+const router = require('express').Router();
+const passport = require('passport');
+let client_add = "http://localhost:3000";
+
+// function ensureAuthenticated
+// ensures the user is logged in before it grants access to the api.
+// input: req, the request; res, the response obj; next, the function to be called next
+function ensureAuthenticated(req, res, next)
+{
+    if (req.isAuthenticated()) {
+        return next();
+    } else {
+        res.status(403).json( {"failure": "you are not logged in"});
+    }
+}
+
+// auth login
+router.get('/login', (req,res) => {
+  res.send('Success');
+});
+
+// auth logout
+router.get('/logout', ensureAuthenticated, (req,res) => {
+    req.logout();
+    // TODO: detect whether logout is successful
+    res.send('Success');
+  // handle with passport
+});
+
+// auth with our own Login
+router.get('/local', (req, res) => {
+    res.sent("Error: GET method not supported.");
+});
+
+router.post('/local', passport.authenticate('local'), (req, res) => {
+    res.json({
```

Comment:
redirect to route handler for GET user, there's alot of logic there that this doesn't encompass